examples/querying_entity_versions.ipynb

```diff
@@ -0,0 +1,782 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "1c0683dd-c601-41b4-a069-d5acb3c40cbf",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "> **Result:**\n",
+       "> `latest`"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "helpers.display_result(version_trait.getSpecifiedTag())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "17049575-fc94-4ac1-9db4-466cccbe1061",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "BAL uses the `v=` query param to define versions in its references. Ours did not have one (`\"bal:///project_artwork/logos/openassetio\"`), in this case, BAL elects to return the latest version. Again note, this is a BAL specific behaviour, other managers may handle things differently.\n",
+    "\n",
+    "The version trait defines that the `specifiedTag` property should hold the tag string that could be used to achieve the same result, so is set to `\"latest\"`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "52fe1001-166a-4356-b6ba-869e76bdd05d",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "> **Result:**\n",
+       "> `2`"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "helpers.display_result(version_trait.getStableTag())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6485867c-086a-4e97-98b6-c56a08da31ef",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "The `stableTag` proptery holds the version tag for the targeted entity that is guaranteed to be free of dynamic behaviour. So in this case, indicates that it is version \"2\".\n",
+    "\n",
+    "Note that not all managers version all entities, this trait will not be imbued to the resulting data if the queried entity is un-versioned."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "802716be-296d-453a-81b2-e068e5a8fc16",
+   "metadata": {
+    "editable": true,
+    "jp-MarkdownHeadingCollapsed": true,
+    "raw_mimetype": "",
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "## Querying other versions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aa250993-ba14-4e3a-be3b-3a2cc0234c30",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "It was noted earlier that Hosts, generally speaking, should never \"string munge\" an existing entity reference to derive another one as this makes assumptions about how any manager may work. This is guaranteed to be different for each manager. \n",
```

Comment:
[Consider] "Guarenteed" is hyperbolic right? I'd cut that final sentence as it dosen't add any additional information.  

```suggestion
    "It was noted earlier that Hosts, generally speaking, should never \"string munge\" an existing entity reference to derive another one as this makes assumptions about how any manager may work. The composition of entity reference strings is certain to be unique to each manager and must not be relied upon. \n",
```
may be more helpful as it provides context on what the problem being solved here is.